kotlinx-coroutines-core/common/src/EventLoop.common.kt

```diff
@@ -410,6 +415,7 @@ internal abstract class EventLoopImplBase: EventLoopImplPlatform(), Delay {
          */
         @JvmField var nanoTime: Long
     ) : Runnable, Comparable<DelayedTask>, DisposableHandle, ThreadSafeHeapNode {
+        @Volatile
```

Comment:
Without this `volatile`, `removeAtImpl` may see an arbitrary value at `assert { result.heap === this }`, leading to a potential crash.